qutip/tests/core/test_qobjevo.py

```diff
@@ -42,111 +42,442 @@
+        obj2 = self.qobjevos[extra_form]
+        obj2_t = self.qobjevos[extra_form](t)
+
+        as_qevo = bin_op(obj1, obj2)(t)
+        as_qobj = bin_op(obj1_t, obj2_t)
+        _assert_qobj_almost_eq(as_qevo, as_qobj)
+
+    @pytest.mark.parametrize('bin_op', [_add, _sub, _mul, _matmul, _tensor],
+                             ids=['add', 'sub', 'mul', 'matmul', 'tensor'])
+    def test_binopt_qobj(self, form, bin_op):
+        "QobjEvo arithmetic"
+        t = self._rand_t()
+        obj = self.qobjevos[form]
+        obj_t = self.qobjevos[form](t)
+        qobj = rand_herm(self.N)
+
+        as_qevo = bin_op(obj, qobj)(t)
+        as_qobj = bin_op(obj_t, qobj)
+        _assert_qobj_almost_eq(as_qevo, as_qobj)
+
+        as_qevo = bin_op(qobj, obj)(t)
+        as_qobj = bin_op(qobj, obj_t)
+        _assert_qobj_almost_eq(as_qevo, as_qobj)
+
+    @pytest.mark.parametrize('bin_op', [_add, _sub, _mul, _div],
+                             ids=['add', 'sub', 'mul', 'div'])
+    def test_binopt_scalar(self, form, bin_op):
+        "QobjEvo arithmetic"
+        t = self._rand_t()
+        obj = self.qobjevos[form]
+        obj_t = self.qobjevos[form](t)
+        scalar = np.random.rand() + 1j
+
+        as_qevo = bin_op(obj, scalar)(t)
+        as_qobj = bin_op(obj_t, scalar)
+        _assert_qobj_almost_eq(as_qevo, as_qobj)
+
+        if bin_op is not _div:
+            as_qevo = bin_op(scalar, obj)(t)
+            as_qobj = bin_op(scalar, obj_t)
+            _assert_qobj_almost_eq(as_qevo, as_qobj)
+
+    @pytest.mark.parametrize('unary_op', [_neg, _dag, _conj, _trans, _cdc],
+                             ids=['neg', 'dag', 'conj', 'trans', '_cdc'])
+    def test_unary(self, form, extra_form, unary_op):
+        "QobjEvo arithmetic"
+        t = self._rand_t()
+        obj = self.qobjevos[form]
+        obj_t = self.qobjevos[form](t)
+
+        as_qevo = unary_op(obj)(t)
+        as_qobj = unary_op(obj_t)
+        _assert_qobj_almost_eq(as_qevo, as_qobj)
+
+    def test_tidyup(self):
+        "QobjEvo tidyup"
+        obj = self.qobjevos["no_args"]
+        obj *= 1e-10 * np.random.random()
+        obj.tidyup(atol=1e-8)
+        t = self._rand_t()
+        # check that the Qobj are cleaned
+        assert_equal(np.max(obj(t).full()), 0.)
+
+    def test_QobjEvo_pickle(self, form):
+        "QobjEvo pickle"
+        # used in parallel_map
+        import pickle
+        pickled = pickle.dumps(self.qobjevos[form], -1)
+        recreated = pickle.loads(pickled)
+        _assert_qobjevo_equivalent(recreated, self.qobjevos[form])
+
+    @pytest.mark.parametrize('superop', [lindblad_dissipator,
+                                         partial(lindblad_dissipator, chi=0.5),
+                                         sprepost,
+                                         lambda O1, O2: liouvillian(O1, [O2])],
+                             ids=['lindblad', 'lindblad_chi',
+                                  'sprepost', 'liouvillian'])
```

Comment:
you can write
```python
pytest.mark.parametrize('superop', [
  pytest.param(lindblad_dissipator, id='lindblad'),
  pytest.param(partial(lindblad_dissipator, chi=0.5), id='lindblad_chi'),
  ...
])
```
to make it easier to read the ids, and to make changes/add lines later.